cli/src/generate/nfa.rs

```diff
@@ -445,27 +448,28 @@ impl<'a> NfaCursor<'a> {
         Self::group_transitions(self.raw_transitions())
     }
 
-    fn raw_transitions(&self) -> impl Iterator<Item = (&CharacterSet, bool, i32, u32)> {
+    fn raw_transitions(&self) -> impl Iterator<Item = (&CharacterSet, bool, i32, u32, bool)> {
```

Comment:
I don't think `eof` should be modeled as part of an `Advance` action. I would think that it would just be a property of an `AcceptToken` action: "this token can only be accepted in this state if we are at EOF".

EOF seems different than a lookahead character set in that it's not something in the input stream that we *consume*; it's just a boolean property (are we or are we not at EOF) of a point in the input stream.

Reply:
I was a bit unsure about how to model the eof action, yeah, there's no actual consumption of a character but we would still want to "advance" to the state when an eof token was used I'd think. Or, we can just accept in place in the zeroth state, is that what you were thinking? Here's my interpretation of it just being an `AcceptToken` action

```c
if (!*accepted_eof && eof) {
        *accepted_eof = true;
        ACCEPT_TOKEN(aux_sym_eof);
		END_STATE();
      }
```

So, instead of advancing to say state 4 and accepting `aux_sym_eof` there, just accept in place in state 0 instead